scripts/broad_dsde_workflows/GenotypeGVCFsScatterWf_170204.wdl

```diff
@@ -0,0 +1,173 @@
+	    File gvcf_index = "${unzipped_basename}.g.vcf.idx"
+    }
+}
+
+# Perform joint-genotyping
+task GenotypeGVCFs { 
+	Array[File] gvcfs
+    Array[File] gvcf_indices
+    String vcf_basename
+    File ref_dict
+    File ref_fasta
+    File ref_fasta_index
+  	File interval_list
+    Int disk_size
+    String mem_size
+
+	command {
+		java -XX:GCTimeLimit=50 -XX:GCHeapFreeLimit=10 -Xmx8000m \
+        	-jar /usr/gitc/GATK36.jar \
+        	-T GenotypeGVCFs \
+        	-R ${ref_fasta} \
+        	--variant ${sep=' --variant ' gvcfs} \
+        	-L ${interval_list} \
+        	-o ${vcf_basename}.vcf.gz 
+	}
+
+	output {
+		File genotyped_vcf = "${vcf_basename}.vcf.gz"
+		File genotyped_index = "${vcf_basename}.vcf.gz.tbi"
+	}
+
+	runtime {
+		docker: "broadinstitute/genomes-in-the-cloud:2.2.4-1469632282"
+		memory: mem_size
+    	cpu: "1"
+    	disks: "local-disk " + disk_size + " HDD"
+	}
+}
+
+# Combine multiple VCFs 
+task MergeVCFs {
+    Array [File] input_vcfs
+    Array [File] input_vcfs_indices
+    String vcf_name
+    String vcf_index
+    Int disk_size
+    String mem_size
+
+    command {
+	    java -Xmx2g -jar /usr/gitc/picard.jar \
+	    MergeVcfs \
+	    INPUT=${sep=' INPUT=' input_vcfs} \
+	    OUTPUT=${vcf_name}
+    }
+
+  	runtime {
+	    docker: "broadinstitute/genomes-in-the-cloud:2.2.4-1469632282"
+	    memory: mem_size
+	    disks: "local-disk " + disk_size + " HDD"
+	}
+
+    output {
+    	File output_vcf = "${vcf_name}"
+    	File output_vcf_index = "${vcf_index}"
+    }
+}
+
+workflow GenotypeGVCFsScatterWf {
+  	File ref_fasta
+  	File ref_fasta_index
+  	File ref_dict
+	Array[File] input_gvcfs
+	Array[File] input_gvcf_indices
+	String cohort_vcf_name
+    File scattered_calling_intervals_list
+  
+    Array[File] scattered_calling_intervals = read_lines(scattered_calling_intervals_list)
+
+    # Unzip GVCFs in parallel
+	scatter (input_gvcf in input_gvcfs) {
+
+		# Unzip GVCFs because GenotypeGVCFs is a cranky toddler
+	    call UnzipGVCF {
+		    input:
+			    gzipped_gvcf = input_gvcf,
+			    unzipped_basename = "temp_unzipped"
+	    }
+	}
+
+	# Joint-genotype variants in parallel over WGS calling intervals
+  	scatter (subInterval in scattered_calling_intervals) {
+
+  		# Perform joint genotyping per interval
+		call GenotypeGVCFs {
+			input:
+			    gvcfs = UnzipGVCF.unzipped_gvcf,
+			    gvcf_indices = UnzipGVCF.gvcf_index,
+			    vcf_basename = cohort_vcf_name,
+			    ref_dict = ref_dict,
+			    ref_fasta = ref_fasta,
+			    ref_fasta_index = ref_fasta_index,
+			    interval_list = subInterval
+		}
+	}
+
+	# Merge per-interval VCFs into a single cohort VCF file
+    call MergeVCFs {
+	    input:
+		    input_vcfs = GenotypeGVCFs.genotyped_vcf,
+		    input_vcfs_indices = GenotypeGVCFs.genotyped_index,
+		    vcf_name = cohort_vcf_name + ".vcf.gz",
+		    vcf_index = cohort_vcf_name + ".vcf.gz.tbi"
+    }
+
+  	# Outputs that will be retained when execution is complete
```

Comment:
I'd rather continue to distinguish the workflow (the thing that gets executed) from its execution.